test/e2e/create_test.go

```diff
@@ -42,64 +43,45 @@ func creationTestSuite(t *testing.T) {
 
 // testWindowsNodeCreation tests the Windows node creation in the cluster
 func testWindowsNodeCreation(t *testing.T) {
-	testCases := []struct {
-		testCase  string
-		isWindows bool
-		replicas  int32
-	}{
-		{
-			// We need to create Windows MachineSet without label first, as we are using waitForWindowsNodes method.
-			testCase:  "Create a Windows MachineSet without label",
-			isWindows: false,
-			replicas:  1,
-		},
-		{
-			testCase:  "Create a Windows MachineSet with label",
-			isWindows: true,
-			replicas:  gc.numberOfNodes,
-		},
-	}
 	testCtx, err := NewTestContext(t)
 	require.NoError(t, err)
 	// Create a private key secret with a randomly generated value. This is changed to the provided private key later in
 	// the test.
 	require.NoError(t, testCtx.createPrivateKeySecret(false), "could not create private key secret")
 
-	for _, test := range testCases {
-		if err := testCtx.createWindowsMachineSet(test.replicas, test.isWindows); err != nil {
-			t.Fatalf("failed to create Windows MachineSet %v ", err)
-		}
-		if test.isWindows {
-			// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
-			// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
-			// provisioned and then change the key. The the correct amount of nodes being configured is proof that the
-			// mismatched Machine created with the mismatched key was deleted and replaced.
-			// Depending on timing and configuration flakes this will either cause all Machines, or all Machines after
-			// the first configured Machines to hit this scenario.
-			err := testCtx.waitForWindowsMachines(int(test.replicas), "Provisioned")
-			require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-			err = testCtx.createPrivateKeySecret(true)
-			require.NoError(t, err, "error replacing private key secret")
-
-		}
-		err := testCtx.waitForWindowsNodes(test.replicas, true, !test.isWindows, false)
+	t.Run("Windows Machines without the Windows label are not configured", func(t *testing.T) {
+		ms, err := testCtx.createWindowsMachineSet(1, false)
+		require.NoError(t, err, "failed to create Windows MachineSet")
+		defer framework.Global.Client.Delete(context.TODO(), ms)
+		err = testCtx.waitForWindowsNodes(1, true, false, false)
```

Comment:
If we are expecting an error here, shouldn`t this be testCtx.waitForWindowsNodes(1, true, true, false)

Reply:
That option reduces the amount of time the function waits for the Windows node. Its not right to use it in this case we want to wait as long as we usually do to ensure that no node is configured. That option really should only be used when deleting nodes.